### ISSUE TITLE
Lint/NestedMethodDefinition shouldn't flag methods defined on Structs

### ISSUE THREAD

**Issue by emschwar** [bbatsov/rubocop#3048]
I have some code that generates a Struct that has some methods added to it:

```ruby
def listener
  @listener ||= Struct.new(:thing, :to, :listen_to) do
    def all_the_things
      "#{thing} #{to} #{listen_to}"
    end
  end
end
```

According to http://ruby-doc.org/core-2.2.0/Struct.html, this is the recommended way to customize a struct, but this code fails the cop:

```
$ bundle exec rubocop -D test.rb
Inspecting 1 file
W

Offenses:

test.rb:3:5: W: Lint/NestedMethodDefinition: Method definitions must not be nested. Use lambda instead.
    def all_the_things ...
    ^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected
```

## Steps to reproduce the problem

The method above should trigger the cop.

## RuboCop version

Include the output of `rubocop -V`:

```
$ rubocop -V
0.39.0 (using Parser 2.3.0.7, running on ruby 2.2.4 x86_64-linux-musl)
```
